src/components/App.tsx

```diff
@@ -430,55 +430,59 @@ class App extends React.Component<ExcalidrawProps, AppState> {
           "excalidraw--view-mode": viewModeEnabled,
         })}
         ref={this.excalidrawContainerRef}
-        style={{
-          width: canvasDOMWidth,
-          height: canvasDOMHeight,
-          top: offsetTop,
-          left: offsetLeft,
-        }}
       >
-        <LayerUI
-          canvas={this.canvas}
-          appState={this.state}
-          setAppState={this.setAppState}
-          actionManager={this.actionManager}
-          elements={this.scene.getElements()}
-          onCollabButtonClick={onCollabButtonClick}
-          onLockToggle={this.toggleLock}
-          onInsertElements={(elements) =>
-            this.addElementsFromPasteOrLibrary(
-              elements,
-              DEFAULT_PASTE_X,
-              DEFAULT_PASTE_Y,
-            )
-          }
-          zenModeEnabled={zenModeEnabled}
-          toggleZenMode={this.toggleZenMode}
-          langCode={getLanguage().code}
-          isCollaborating={this.props.isCollaborating || false}
-          onExportToBackend={onExportToBackend}
-          renderCustomFooter={renderFooter}
-          viewModeEnabled={viewModeEnabled}
-          showExitZenModeBtn={
-            typeof this.props?.zenModeEnabled === "undefined" && zenModeEnabled
-          }
-        />
-        <div className="excalidraw-textEditorContainer" />
-        {this.state.showStats && (
-          <Stats
+        <div
+          style={{
+            width: canvasDOMWidth,
+            height: canvasDOMHeight,
+            top: offsetTop,
+            left: offsetLeft,
+          }}
+        >
```

Comment:
from my testing this is actually not ideal. What works best if dropping this `div`, and putting just `width`/`height` on the `.excalidraw` (while keeping the rest of the changes in this PR).

Reply:
The above *only* applies when user doesn't supply `offsetLeft`/`offsetTop`.

Reply:
Maybe we should drop support for `offsetLeft`/`offsetTop` with this PR, and make the host apps control the positioning by wrapping the component and position the container.

Reply:
The `offsetLeft`/`offsetTop` made sense only when Excalidraw was `fixed`-positioned anyway.

Reply:
The problem with the container solution is it'd be hard for us to detect position change.

I think we should expose an API to recalculate offsets for now, before we figure out how to automate it (in a performant way). The solution with supplying custom offsets isn't good: e.g. when you want to flex-align the Excal component.

Reply:
yep, agreed we can remove it.

Reply:
>The problem with the container solution is it'd be hard for us to detect position change.
I think we should expose an API to recalculate offsets for now, before we figure out how to automate it (in a performant way). The solution with supplying custom offsets isn't good: e.g. when you want to flex-align the Excal component.

didn't get you, are you talking about the offset props which we have right now which host uses when having cases like some collapsable pane which will collapse/expand excalidraw ? This I was thinking we can solve using resize observer API.

Reply:
If the excalidraw container (we have control over) resizes, then that could be solved via ResizeObserver. But if it just changes position, that won't work :(.

Reply:
I need a bit more clarification on the use case, when will there be such a use case that just position changes and not width/height?  I might be missing something. By exposing API you mean we just expose the API where we calculate the offsets from the parent container right ?
But this can be done in a separate PR since this is a patch release and exposing an API would be a minor release.

Reply:
> I need a bit more clarification on the use case, when will there be such a use case that just position changes and not width/height? I might be missing something.
 
let's say the component is inside a flex container that's x-centering its children. The container may get smaller and thus the x-center position will change, without necessarily changing the dimension of the excal component (there may be a lot of buffer space). That's just one case.

> By exposing API you mean we just expose the API where we calculate the offsets from the parent container right ?

Yep. Let's leave for future PRs.